runtime/criusupport/criusupport.cpp

```diff
@@ -611,6 +611,13 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 			goto wakeJavaThreads;
 		}
 
+		/* Single thread mode needs to be turned off before delayed lock related
+		 * events are sent as a thread which is in a wait() call it will attempt to
+		 * re-acquire the monitor after it is notified. Should that happen in single
+		 * thread mode an exception will be thrown.
+		 */
+		vm->extendedRuntimeFlags2 &= ~J9_EXTENDED_RUNTIME2_CRIU_SINGLE_THREAD_MODE;
```

Comment:
Having released exclusive, there may be an issue here modifying this field without lock.

Reply:
Are the non-current threads still suspended at this point?

Reply:
Yes all java threads are still suspended. We dont have a lock for extendedRuntimes flags and I dont think other callers will lock on it. I've moved it to checkpointState, which guarantees it will be only written by the checkpoint thread.